cas-server-support-osf/src/main/java/io/cos/cas/authentication/handler/support/OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction.java

```diff
@@ -408,6 +453,35 @@ private JSONObject normalizeRemotePrincipal(final OpenScienceFrameworkCredential
         return XML.toJSONObject(writer.getBuffer().toString());
     }
 
+    /**
+     * Construct normalized payload from remote principal authenticated by external CAS.
+     *
+     * @param principal the principal created by external CAS authentication
+     * @return the json object to serialize for authorization with the OSF API
+     */
+    private JSONObject constructPayloadFromRemotePrincipal(final Principal principal) {
```

Comment:
Although OKState uses SAML11 for CAS service validation, `pac4j` handles the parsing  of the XML response. Maybe this is the only case, but a generalized solution is preferred.